x-pack/plugins/ingest_manager/server/plugin.ts

```diff
@@ -115,12 +136,14 @@ export class IngestManagerPlugin
   private isProductionMode: boolean;
   private kibanaVersion: string;
   private httpSetup: HttpServiceSetup | undefined;
+  private externalCallbacks: ExternalCallbacksStorage;
 
   constructor(private readonly initializerContext: PluginInitializerContext) {
     this.config$ = this.initializerContext.config.create<IngestManagerConfigType>();
     this.isProductionMode = this.initializerContext.env.mode.prod;
     this.kibanaVersion = this.initializerContext.env.packageInfo.version;
     this.logger = this.initializerContext.logger.get();
+    this.externalCallbacks = new Map();
```

Comment:
I am not sure this state should be managed in the class, as it can be seen there was no unit test.

Reply:
@nnamdifrankie are you suggesting instead that we only store it in `appContextService` ? (I think that is what you're saying 😄  - please confirm

Reply:
So this is the plugin class and is responsibly for initialization. But I am fine if everyone is fine with it including the map storage here too.

Reply:
I can move it off the the app context and keep it out of the Plugin interface